More PPU debugger updates

undisbeliever opened the pull request:
This pull request updates the Tilemap Viewer and adds a new window, Tile Viewer, to the debugger.

<br/>

Changes to the Tilemap Viewer:
 * Add an Export button to save the tilemap as a PNG image ([suggested by qwertymodo](https://github.com/devinacker/bsnes-plus/issues/143))
 * Allow user to override the background color ([suggested by qwertymodo](https://github.com/devinacker/bsnes-plus/issues/143))

<br/>

The Tile Viewer allows the user to view tiles within VRAM and the CPU bus. It uses ImageGridWidget to draw the tiles and the same tile rendering code as the Tilemap Viewer (as suggested by @devinacker some time ago).


Notable features of Tile Viewer include:
 * Export button to save tiles as PNG Image
 * The user can override the background color
 * User adjustable width ([suggested by qwertymodo](https://github.com/devinacker/bsnes-plus/issues/62#issuecomment-257656033))
 * "goto" buttons set the bit-depth when clicked
    * The goto buttons do not set the Bit Depth in mode 7 as the game could be changing the screen modes mid frame
 * Next/Previous address buttons to allow for easier tile searching

![zelda-text](https://user-images.githubusercontent.com/65687/32265965-145febe4-bf31-11e7-97c2-70579a82837a.png)

If everyone is happy with the Tile Viewer we can retire the old Vram Viewer window. 

---
Paging @devinacker: Is [this](https://github.com/devinacker/bsnes-plus/blob/cc38bdf81ab2bf8a9df3cf66c65561b49ddcfd37/bsnes/ui-qt/debugger/ppu/tile-renderer.cpp#L145-L161) the correct way to access the CPU bus?

<br/>

Paging @Optiroc: When you encounter merge conflicts, use my code instead of yours.

I was unable to test this push request with optiroc/master as there is no mingw build of QtWebEngine ([source](https://doc.qt.io/qt-5/qtwebengine-platform-notes.html#windows)). Instead, I tested the code with commit e24f81e on Windows 7 Qt 5.9.0 x86-mingw32 and Qt 5.9.2 Archlinux x86_64

Optiroc commented:
Nice! Being able to view S-CPU bus memory as tile data is already proving extremely useful in my [current project](http://www.romhacking.net/forum/index.php?topic=25186.0)!

devinacker commented:
> Paging @devinacker: Is this the correct way to access the CPU bus?

You can also call `SNES::bus.read` directly, but either one works.

I'll do some more testing/code review shortly.

devinacker commented:
... and by "shortly", I mean a month later, apparently. (Still haven't looked at the actual code though)

Anyway, looks good to me so far. My only suggestion right now is adding "cartridge ROM" as a third source.

undisbeliever commented:
Added Mode7 EXTBG format to the PPU Debugger:
![Contra 3 Stage 2](https://user-images.githubusercontent.com/65687/39698101-bec6e8e2-5236-11e8-89f2-1753bef868ed.png)
*Source: Contra 3 - Stage 2*


Added more sources to the Tile Viewer
![Yoshi's Island](https://user-images.githubusercontent.com/65687/39698114-ca081f64-5236-11e8-9689-e6a095307afb.png)
*Source: Yoshi's Island*

---

I think Tile Viewer is now feature complete.

Paging @devinacker: Could I please get a status report on this pull request?

devinacker commented:
I'll look at those latest commits today. I'll be working on more stuff for the v4 release this week so this should be merged soon, possibly also today if everything looks good.

devinacker commented:
Alright, the newest commits look good to me so far. The default width of the tile viewer could probably be expanded horizontally to fit the entire view, but that's a minor nitpick. I'll see about finally getting this stuff merged tomorrow and phasing out the old VRAM viewer.
